cmd/validator/validator.go

```diff
@@ -80,15 +85,31 @@ func getFlags() (validatorConfig, error) {
 	versionPtr := flag.Bool("version", false, "Version prints the release version of validator")
 	groupOutputPtr := flag.String("groupby", "", "Group output by filetype, directory, pass-fail. Supported for Standard and JSON reports")
 	quietPrt := flag.Bool("quiet", false, "If quiet flag is set. It doesn't print any output to stdout.")
+	globbingPrt := flag.Bool("globbing", false, "If globbing flag is set, check for glob patterns in the arguments.")
```

Comment:
Not sure how I feel about the globbing flag - I can't think of another CLI that requires a flag to enable globbing patterns. Can you help me understand the potential impacts of globbing being implicit?

Reply:
When globbing is implicit, it might be harder for users to understand why certain files were included or excluded in the tool’s operation. 

I’m also unsure how globbing should interact with other flags. Testing it, I found that the `-exclude-dirs`, does not work 100% with my implementation of globbing. Specifically, if the directory to exclude is in the middle of the path matched. Do you think the `-exclude-dirs` should take priority over the globbing pattern matching? I can implement that as well.

For example:
`validator -exclude-dirs=subdir2 -globbing "test/**"` would exclude subdir2.
`validator -exclude-dirs=subdir2 -globbing "test/**/*"` would not exclude subdir2.
`validator -exclude-dirs=fixtures -globbing "test/**/"` would not exclude fixtures.

The `-exclude-file-types` flag currently takes priority. For example, `validator -exclude-file-types=json "test/**/*.json"` would not return any results.

These felt like potentially breaking changes to me, so I decided to keep this feature under a flag. If you’re okay enabling it by default, let me know.

Reply:
Okay good thoughts, I'm convinced on the merits of the `--globbing` flag. I think you're right that it needs to be standalone and not used with `--exclude-dirs` or `--exclude-file-types`. I think we should enforce that so if you choose globbing, that's the only way you get to filter the files. What do you think?

Reply:
@kehoecj sounds good! I'll work on restricting the flag usage and refactor to pass the go report card.